### ISSUE TITLE
About training: different results.

### ISSUE THREAD

**Issue by ChrisChenYh** [dqj5182/CONTHO_RELEASE#8]
Hello, many thanks for your code sharing.
I'm now trying to reproduce the training process on BEHAVE (download from official website). I keep the same settings as the initial code. However, the final result is quite defferent from your paper Table 4. Following are my config and training log. I wonder if you have some advice.

`08-06 00:25:29 Work on GPU: 2
08-06 00:25:29 Args: Namespace(resume_training=False, gpu='2', dataset='behave', exp='', checkpoint='')
08-06 00:25:29 Cfg: {'cur_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core', 'root_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../', 'data_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../data', 'DATASET': {'name': 'BEHAVE', 'workers': 4, 'random_seed': 123, 'bbox_expand_ratio': 1.3, 'obj_set': 'behave'}, 'MODEL': {'input_img_shape': [512, 512], 'input_body_shape': [256, 256], 'input_hand_shape': [256, 256], 'img_feat_shape': [8, 8, 8], 'weight_path': ''}, 'TRAIN': {'batch_size': 16, 'shuffle': True, 'begin_epoch': 1, 'end_epoch': 50, 'warmup_epoch': 3, 'scheduler': 'step', 'lr': 0.0001, 'min_lr': 1e-06, 'lr_step': [30], 'lr_factor': 0.1, 'optimizer': 'adam', 'momentum': 0, 'weight_decay': 0, 'beta1': 0.5, 'beta2': 0.999, 'print_freq': 10, 'loss_names': ['contact', 'vert', 'edge', 'param', 'coord', 'hand_bbox'], 'contact_loss_weight': 1.0, 'smpl_vert_loss_weight': 1.0, 'obj_vert_loss_weight': 1.0, 'smpl_edge_loss_weight': 1.0, 'smpl_pose_loss_weight': 1.0, 'smpl_shape_loss_weight': 1.0, 'obj_pose_loss_weight': 1.0, 'obj_trans_loss_weight': 1.0, 'smpl_3dkp_loss_weight': 1.0, 'smpl_2dkp_loss_weight': 1.0, 'pos_2dkp_loss_weight': 1.0, 'hand_bbox_loss_weight': 1.0}, 'AUG': {'scale_factor': 0.2, 'rot_factor': 30, 'shift_factor': 0, 'color_factor': 0.2, 'blur_factor': 0, 'flip': False}, 'TEST': {'batch_size': 1, 'shuffle': False, 'do_eval': True, 'eval_metrics': ['contact_est_p', 'contact_est_r', 'cd_human', 'cd_object', 'contact_rec_p', 'contact_rec_r'], 'print_freq': 10, 'contact_thres': 0.05}, 'CAMERA': {'focal': [1000, 1000], 'princpt': [256.0, 256.0], 'depth_factor': 4.4, 'obj_depth_factor': 4.4}, 'output_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29', 'graph_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29/graph', 'vis_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29/vis', 'res_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29/results', 'log_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29/log', 'checkpoint_dir': '/root/repo/chenyuhang/CONTHO_RELEASE/lib/core/../../experiment/exp_08-06_01:25:29/checkpoints'}
08-06 00:25:40 # of model parameters: 82802022
08-06 00:25:40 ==> Preparing TRAIN Dataloader...
08-06 00:26:04 # of TRAIN BEHAVE data: 45380
08-06 00:26:04 # of model parameters: 82802022
08-06 00:26:04 ==> Preparing TEST Dataloader...
08-06 00:26:10 # of TEST BEHAVE data: 4129
08-06 00:26:10 ===> Start training...
08-06 01:01:53 Epoch1 Loss: total: 0.1358 contact: 0.0295 vert: 0.0179 edge: 0.0003 param: 0.0674 coord: 0.0182 hand_bbox: 0.0025
08-06 01:22:08 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.676 / Recall: 0.099
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 10.15 / Object: 21.64
>> Contact from reconstruction
Precision: 0.424 / Recall: 0.242
08-06 01:51:17 Epoch2 Loss: total: 0.1023 contact: 0.0252 vert: 0.0126 edge: 0.0003 param: 0.0466 coord: 0.0159 hand_bbox: 0.0017
08-06 02:20:09 Epoch3 Loss: total: 0.0918 contact: 0.0238 vert: 0.0115 edge: 0.0003 param: 0.0394 coord: 0.0154 hand_bbox: 0.0015
08-06 02:49:01 Epoch4 Loss: total: 0.0866 contact: 0.0229 vert: 0.0108 edge: 0.0003 param: 0.0361 coord: 0.0151 hand_bbox: 0.0014
08-06 03:17:56 Epoch5 Loss: total: 0.0826 contact: 0.0222 vert: 0.0101 edge: 0.0003 param: 0.0338 coord: 0.0149 hand_bbox: 0.0014
08-06 03:46:39 Epoch6 Loss: total: 0.0796 contact: 0.0216 vert: 0.0096 edge: 0.0003 param: 0.0320 coord: 0.0148 hand_bbox: 0.0013
08-06 04:15:42 Epoch7 Loss: total: 0.0771 contact: 0.0211 vert: 0.0092 edge: 0.0003 param: 0.0306 coord: 0.0147 hand_bbox: 0.0013
08-06 04:44:39 Epoch8 Loss: total: 0.0752 contact: 0.0207 vert: 0.0088 edge: 0.0003 param: 0.0295 coord: 0.0146 hand_bbox: 0.0012
08-06 05:13:29 Epoch9 Loss: total: 0.0736 contact: 0.0203 vert: 0.0086 edge: 0.0003 param: 0.0286 coord: 0.0145 hand_bbox: 0.0012
08-06 05:42:29 Epoch10 Loss: total: 0.0721 contact: 0.0200 vert: 0.0083 edge: 0.0003 param: 0.0278 coord: 0.0145 hand_bbox: 0.0012
08-06 06:11:27 Epoch11 Loss: total: 0.0708 contact: 0.0197 vert: 0.0081 edge: 0.0003 param: 0.0271 coord: 0.0144 hand_bbox: 0.0012
08-06 06:29:53 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.746 / Recall: 0.512
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 6.86 / Object: 12.98
>> Contact from reconstruction
Precision: 0.520 / Recall: 0.436
08-06 06:58:31 Epoch12 Loss: total: 0.0696 contact: 0.0194 vert: 0.0079 edge: 0.0003 param: 0.0265 coord: 0.0143 hand_bbox: 0.0012
08-06 07:27:12 Epoch13 Loss: total: 0.0684 contact: 0.0191 vert: 0.0077 edge: 0.0003 param: 0.0259 coord: 0.0143 hand_bbox: 0.0011
08-06 07:55:56 Epoch14 Loss: total: 0.0674 contact: 0.0189 vert: 0.0075 edge: 0.0003 param: 0.0254 coord: 0.0142 hand_bbox: 0.0011
08-06 08:24:18 Epoch15 Loss: total: 0.0664 contact: 0.0186 vert: 0.0073 edge: 0.0003 param: 0.0249 coord: 0.0142 hand_bbox: 0.0011
08-06 08:52:59 Epoch16 Loss: total: 0.0656 contact: 0.0184 vert: 0.0072 edge: 0.0002 param: 0.0245 coord: 0.0142 hand_bbox: 0.0011
08-06 09:21:41 Epoch17 Loss: total: 0.0647 contact: 0.0182 vert: 0.0070 edge: 0.0002 param: 0.0240 coord: 0.0141 hand_bbox: 0.0011
08-06 09:50:07 Epoch18 Loss: total: 0.0639 contact: 0.0179 vert: 0.0069 edge: 0.0002 param: 0.0236 coord: 0.0141 hand_bbox: 0.0011
08-06 10:18:43 Epoch19 Loss: total: 0.0631 contact: 0.0177 vert: 0.0068 edge: 0.0002 param: 0.0232 coord: 0.0141 hand_bbox: 0.0011
08-06 10:47:11 Epoch20 Loss: total: 0.0624 contact: 0.0176 vert: 0.0067 edge: 0.0002 param: 0.0229 coord: 0.0141 hand_bbox: 0.0011
08-06 11:15:31 Epoch21 Loss: total: 0.0618 contact: 0.0174 vert: 0.0066 edge: 0.0002 param: 0.0225 coord: 0.0140 hand_bbox: 0.0010
08-06 11:33:43 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.744 / Recall: 0.570
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 6.40 / Object: 12.00
>> Contact from reconstruction
Precision: 0.595 / Recall: 0.403
08-06 12:02:38 Epoch22 Loss: total: 0.0611 contact: 0.0172 vert: 0.0065 edge: 0.0002 param: 0.0222 coord: 0.0140 hand_bbox: 0.0010
08-06 12:31:16 Epoch23 Loss: total: 0.0605 contact: 0.0170 vert: 0.0064 edge: 0.0002 param: 0.0219 coord: 0.0140 hand_bbox: 0.0010
08-06 12:59:39 Epoch24 Loss: total: 0.0600 contact: 0.0169 vert: 0.0063 edge: 0.0002 param: 0.0216 coord: 0.0140 hand_bbox: 0.0010
08-06 13:27:41 Epoch25 Loss: total: 0.0594 contact: 0.0167 vert: 0.0062 edge: 0.0002 param: 0.0213 coord: 0.0139 hand_bbox: 0.0010
08-06 13:55:47 Epoch26 Loss: total: 0.0589 contact: 0.0166 vert: 0.0061 edge: 0.0002 param: 0.0210 coord: 0.0139 hand_bbox: 0.0010
08-06 14:23:51 Epoch27 Loss: total: 0.0584 contact: 0.0164 vert: 0.0061 edge: 0.0002 param: 0.0208 coord: 0.0139 hand_bbox: 0.0010
08-06 14:52:07 Epoch28 Loss: total: 0.0579 contact: 0.0163 vert: 0.0060 edge: 0.0002 param: 0.0206 coord: 0.0139 hand_bbox: 0.0010
08-06 15:20:50 Epoch29 Loss: total: 0.0573 contact: 0.0161 vert: 0.0059 edge: 0.0002 param: 0.0203 coord: 0.0139 hand_bbox: 0.0010
08-06 15:49:22 Epoch30 Loss: total: 0.0569 contact: 0.0160 vert: 0.0058 edge: 0.0002 param: 0.0200 coord: 0.0139 hand_bbox: 0.0010
08-06 16:17:54 Epoch31 Loss: total: 0.0565 contact: 0.0159 vert: 0.0058 edge: 0.0002 param: 0.0198 coord: 0.0138 hand_bbox: 0.0010
08-06 16:35:24 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.771 / Recall: 0.543
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 5.95 / Object: 11.49
>> Contact from reconstruction
Precision: 0.621 / Recall: 0.419
08-06 17:03:53 Epoch32 Loss: total: 0.0560 contact: 0.0157 vert: 0.0057 edge: 0.0002 param: 0.0196 coord: 0.0138 hand_bbox: 0.0009
08-06 17:32:21 Epoch33 Loss: total: 0.0556 contact: 0.0156 vert: 0.0057 edge: 0.0002 param: 0.0194 coord: 0.0138 hand_bbox: 0.0009
08-06 18:00:51 Epoch34 Loss: total: 0.0552 contact: 0.0155 vert: 0.0056 edge: 0.0002 param: 0.0192 coord: 0.0138 hand_bbox: 0.0009
08-06 18:29:10 Epoch35 Loss: total: 0.0548 contact: 0.0153 vert: 0.0056 edge: 0.0002 param: 0.0190 coord: 0.0138 hand_bbox: 0.0009
08-06 18:57:27 Epoch36 Loss: total: 0.0544 contact: 0.0152 vert: 0.0055 edge: 0.0002 param: 0.0189 coord: 0.0138 hand_bbox: 0.0009
08-06 19:26:19 Epoch37 Loss: total: 0.0540 contact: 0.0151 vert: 0.0054 edge: 0.0002 param: 0.0187 coord: 0.0138 hand_bbox: 0.0009
08-06 19:55:17 Epoch38 Loss: total: 0.0535 contact: 0.0149 vert: 0.0054 edge: 0.0002 param: 0.0184 coord: 0.0137 hand_bbox: 0.0009
08-06 20:24:03 Epoch39 Loss: total: 0.0532 contact: 0.0148 vert: 0.0053 edge: 0.0002 param: 0.0183 coord: 0.0137 hand_bbox: 0.0009
08-06 20:52:43 Epoch40 Loss: total: 0.0531 contact: 0.0147 vert: 0.0053 edge: 0.0002 param: 0.0182 coord: 0.0138 hand_bbox: 0.0009
08-06 21:21:38 Epoch41 Loss: total: 0.0528 contact: 0.0145 vert: 0.0053 edge: 0.0001 param: 0.0180 coord: 0.0141 hand_bbox: 0.0009
08-06 21:39:20 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.748 / Recall: 0.578
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 5.80 / Object: 10.70
>> Contact from reconstruction
Precision: 0.596 / Recall: 0.477
08-06 22:08:28 Epoch42 Loss: total: 0.0523 contact: 0.0144 vert: 0.0052 edge: 0.0001 param: 0.0178 coord: 0.0140 hand_bbox: 0.0009
08-06 22:37:19 Epoch43 Loss: total: 0.0520 contact: 0.0143 vert: 0.0051 edge: 0.0001 param: 0.0176 coord: 0.0139 hand_bbox: 0.0009
08-06 23:06:04 Epoch44 Loss: total: 0.0517 contact: 0.0142 vert: 0.0051 edge: 0.0001 param: 0.0175 coord: 0.0139 hand_bbox: 0.0009
08-06 23:34:50 Epoch45 Loss: total: 0.0513 contact: 0.0141 vert: 0.0050 edge: 0.0001 param: 0.0173 coord: 0.0139 hand_bbox: 0.0009
08-07 00:03:27 Epoch46 Loss: total: 0.0511 contact: 0.0140 vert: 0.0050 edge: 0.0001 param: 0.0172 coord: 0.0139 hand_bbox: 0.0009
08-07 00:32:09 Epoch47 Loss: total: 0.0508 contact: 0.0139 vert: 0.0050 edge: 0.0001 param: 0.0171 coord: 0.0139 hand_bbox: 0.0009
08-07 01:01:08 Epoch48 Loss: total: 0.0504 contact: 0.0138 vert: 0.0049 edge: 0.0001 param: 0.0169 coord: 0.0139 hand_bbox: 0.0008
08-07 01:30:21 Epoch49 Loss: total: 0.0501 contact: 0.0137 vert: 0.0049 edge: 0.0001 param: 0.0168 coord: 0.0138 hand_bbox: 0.0008
08-07 01:59:22 Epoch50 Loss: total: 0.0498 contact: 0.0135 vert: 0.0048 edge: 0.0001 param: 0.0166 coord: 0.0138 hand_bbox: 0.0008
08-07 02:17:28 Finished Evaluation!
-------- Evaluation Results (Contact estimation) ------
Precision: 0.741 / Recall: 0.592
---------- Evaluation Results (Reconstruction) --------
>> Chamfer Distance
Human: 5.67 / Object: 10.62
>> Contact from reconstruction
Precision: 0.614 / Recall: 0.479`

Looking forward to your reply.

**Comment by faithbotbbot** [comment]
I have the same problem. I keep the training configuration in the code, but the results are quite different from those in the article.

**Comment by dqj5182** [comment]
Before releasing the code, I have trained CONTHO based on this code and achieved similar results with the published paper (not exactly same results due to stochastic property of the training process). So, I am guessing the problem may be due to different Python, PyTroch, and CUDA version. May I ask for you to share the environment setting? For example, which Python, PyTorch and CUDA version are being used.

**Comment by ChrisChenYh** [comment]
![image](https://github.com/user-attachments/assets/4954e3b0-454d-46dd-a5d5-5911afcfae5d)

**Comment by ChrisChenYh** [comment]
Python=3.9.19, PyTorch=1.8.1 and CUDA=11.1

**Comment by dqj5182** [comment]
Could you match the PyTorch version to 1.10.1 and rerun the training? Let me know if the result still fall short of that reported in our paper.

**Comment by ChrisChenYh** [comment]
OK, I will match the evironment settings and rerun the training. Sincerely thanks for your reply!

**Comment by ChrisChenYh** [comment]
> Could you match the PyTorch version to 1.10.1 and rerun the training? Let me know if the results still fall short of that reported in our paper.

I match the pytorch version ot 1.10.1 and rerun the training,  get the correct results. Thank U so much! 👍 ![image](https://github.com/user-attachments/assets/52dcdf1f-5fc6-4f2c-b149-96137cdca9fd)

**Comment by dqj5182** [comment]
No problem!

Glad that the problem has been solved :)
